src/mongoose_lib.erl

```diff
@@ -148,3 +149,26 @@ parse_ip_netmask(IPStr, MaskStr) ->
         _ ->
             error
     end.
+
+%% Runs a function for each element on the same node
+pmap(F, Es) ->
+    pmap(F, Es, 5000).
+
+pmap(F, Es, Timeout) ->
+    Parent = self(),
+    Running = 
+        [spawn_monitor(fun() -> Parent ! {pmap_result, self(), F(E)} end)
+            || E <- Es],
+    collect(Running, Timeout).
```

Comment:
Now here there's a problem. I don't think this is a reason to crash a c2s process, and actually, it won't crash it, because the whole thing is ran in a big try-catch. And if any of those parallel processes running the callback takes longer than the configured time, the main process, in this case most usually the c2s, will receive delayed extraneous messages that he won't know how to handle.

This is precisely what `rpc:pmap` tries to solve: it uses a proxy process (rpc unfortunately uses a single global gen_server, which is an enormous bottleneck, but we can spawn new ones on request here, that spawns all the other processes. If any message was to be delayed, the proxy process wouldn't be alive anymore, so the delayed messages would be silently dropped.

I think that is the correct way to do it: spawn_monitor a single proxy process that will just die in `Timeout` time for sure. That process will do the list of spawn_monitor the parallel workers and do all the collects and answer to the original call.

Note also that the current `collect` function is doing `after Timeout` on every message, so the whole thing could potentially take as long as `Timeout * number-of-workers`. That should also be fixed.

Reply:
it is executed from tests only